### ISSUE TITLE
controlled `inputValue` is overridden when controlled `selectedItem` is cleared

### ISSUE THREAD

**Issue by ilovett** [downshift-js/downshift#627]
- `downshift` version: 3.1.5
- `node` version: 10.12.0
- `npm` (or `yarn`) version: 1.12.3

**Relevant code or config**

```javascript

clearSelection() {
  this.setState({ selectedItem: null });
}

render() {
const { input, selectedItem } = this.state;

// inputvalue is dynamically set by selectedItem object or fallback to input string
let inputValue = input.value;
if (selectedItem) {
  inputValue = selectedItem.description;
}

<Downshift
  selectedItem={ selectedItem }
  inputValue={ inputValue }
/>
}
```

**What you did**:

When the selectedItem is cleared, there is logic to set the internal state inputValue to use a value determined by itemToString on an empty selection:

![image](https://user-images.githubusercontent.com/1776100/49191025-c343a880-f329-11e8-8a39-8eec9664e9e2.png)

The internal state ignores the controlled `inputValue` and dispatches `onInputValueChange` with an empty string...

P.S. I am controlling `inputValue` with `redux-form` and want the form value to be the string in the input OR fully some selectedItem as an object.  The empty string dispatches up to redux-form which thinks the form value should be empty string and then the controlled `inputValue` unintentionally becomes empty

**Suggested solution**:

When clearing `selectedItem` give priority to controlled `inputValue` over determining `itemToString` of falsy selected item.

**Comment by jazzytomato** [comment]
I am having the same issue, for now I have fixed it by downgrading downshift to ^2.0.20 (same as in the code samples which work fine [here](https://codesandbox.io/s/github/kentcdodds/downshift-examples))

With the same code samples, you can see the problem by upgrading downshift to 3.0.15  https://codesandbox.io/s/j5rlpj6yw

The problem is obvious in the `04-multi-select.js` component

**Comment by kentcdodds** [comment]
Ah yeah, `04-multi-select.js` has an obvious problem, but I'm struggling to see how the issue opened by the original poster is related because the `inputValue` is not controlled in that example. Are these related? In any case, that should definitely be fixed. If it's unrelated, we'll want a new issue for it.

As for the original poster's issue, the `internalSetState` method wont actually change internal state for things which are controlled. Could you provide a reproduction of your issue?
